api.md

```diff
@@ -0,0 +1,73 @@
+
+* `/api/institutions/<institution>/years`
+    * `GET` - List of all years that a given institution filed/will file for. Structure: `{"years": ["2017"]}`
+  
+* `/api/years/<year>/institutions/<institution>`
+* `/api/institutions/<institution>/years/<year>`
+    * `GET` - Data for a single institution for a given year. Structure: [institutions](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/user1-institutions.json#L48-L56)
+
+* `/api/years/<year>/institutions/<institution>/submissions`
+* `/api/institutions/<institution>/years/<year>/submissions`
+    * `GET` - List of all submissions for an institution in a given year. Structure: [submissions](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/submissions.json#L26-L45) (without the extraneous key, which is just a convenience used by the dev server)
+    * `POST` - Create a new submission, incrementing the submission count. Return structure: `{"currentSubmission": 4}`
+
+* `/api/years/<year>/institutions/<institution>/submissions/<submission>`
+* `/api/institutions/<institution>/years/<year>/submissions/<submission>`
+    * `GET` - Data of a submission for an institution in a given year. Structure: [submissions](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/submissions.json#L36-L45)
+    * `POST` - Endpoint for upload of LAR file. POST data: LAR file. Return structure: `{"progress": <progress endpoint>}`
+
+* `/api/years/<year>/institutions/<institution>/submissions/<submission>/progress`
+* `/api/institutions/<institution>/years/<year>/submissions/<submission>/progress`
+    * `GET` - Returns information about the progress of validation and submission in general. *This endpoint may support long-polling*. Structure:
+    ```
+    {
+      status: {
+        code: 3,
+        message: ""
+      },
+      editCounts: {
+        syntactical: 0,
+        validity: 1,
+        quality: 8,
+        macro: 0
+      }
+    }
+    ```
+
+* `/api/years/<year>/institutions/<institution>/submissions/<submission>/edits`
+* `/api/institutions/<institution>/years/<year>/submissions/<submission>/edits`
+    * `GET` - List of all edits for a given submission, grouped by edit type. Structure: `{"syntactical": {...}, "validity": {..}, "quality": {...}, "macro": {...}}`
+
+* `/api/years/<year>/institutions/<institution>/submissions/<submission>/edits`
+* `/api/institutions/<institution>/years/<year>/submissions/<submission>/edits`
+    * `GET` - List of all edits for a given submission, grouped by edit type. Structure: `{"syntactical": {...}, "validity": {..}, "quality": {...}, "macro": {...}}`
+    * `PUT` - Bulk update of edits, used for verification - TODO
+
+* `/api/years/<year>/institutions/<institution>/submissions/<submission>/syntactial|validity|quality|macro`
```

Comment:
Seeing it now, I'm not sure I like the idea of offering multiple endpoints for the same data. Seems to add confusion, and could be confusing for others as well. I think one clean path to the endpoint is enough. Same for the other places this exists. Anyone else?